### DISCUSSION TITLE
file_size_byte comparison

### DISCUSSION THREAD

**Discussion by JunseoChoJJ** [trinodb/trino#22043] [Q&A] [answered]
I have about 200mb data 
and I query 

select sum("$file_size")
from hive.logs.test

it gives me 191852695999421

select * from iceberg.logs."test$files"
when I query I get file_size_in_bytes 141393457

is this two query same? it gives me so different file size make me feek awkward
or is it iceberg super cool skill ?

**Comment by findepi** [comment, marked as answer]
For a file of size S that has N rows, the first query sums N*S and the second sums S.

**Reply by JunseoChoJJ** [reply]
@findeapi thanks I really appreciate it

**Comment by JunseoChoJJ** [comment]
@findepi Can I ask you some question new to data engineer world less than one month, still confuse ,,,

my stack is trino+iceberg+s3 on k8s

I have three questions.

First, I add data into existing iceberg table using insert into
and it works but somehow I got another folder(or bucket bucket seems more right) call trino+segments
I don't know why trino+segments came into my obs. 

Second, is it possible to put csv data file into iceberg table. seems like dremio works but not sure in trino
I don't use dremio by the way just curious because I thought I can put only parquet,orc,avro into iceberg table

Last, in spark there is rewrite_manifests and rewrite_datafiles two another method but in trino you just use optimize
so when I query, ALTER TABLE test_table EXECUTE optimize
this thing, it just automatically rewrite my data file and my manifest file?
I use that and seems like my data file compact in a way but does not rewrite manifest file
or should I do expire snapshot first, and then optimize command again?
